go/vt/vtadmin/cluster/cluster.go

```diff
@@ -768,6 +768,116 @@ func (c *Cluster) getShardSets(ctx context.Context, keyspaces []string, keyspace
 	return shardsByKeyspace, nil
 }
 
+// GetCellInfos returns a list of ClusterCellInfo objects for cells in the
+// given cluster.
+//
+// If req.Cells is set, cells are restricted only to cells with those names.
+// Note: specifying a cell name that does not exist in the cluster fails the
+// overall request.
+//
+// If req.NamesOnly is set, each ClusterCellInfo will only contain the Cluster
+// and Name fields. req.Cells takes precedence over this option.
+func (c *Cluster) GetCellInfos(ctx context.Context, req *vtadminpb.GetCellInfosRequest) ([]*vtadminpb.ClusterCellInfo, error) {
+	span, ctx := trace.NewSpan(ctx, "Cluster.GetCellInfos")
+	defer span.Finish()
+
+	if err := c.Vtctld.Dial(ctx); err != nil {
+		return nil, err
+	}
+
+	names := req.Cells
+	if len(names) == 0 {
+		resp, err := c.Vtctld.GetCellInfoNames(ctx, &vtctldatapb.GetCellInfoNamesRequest{})
+		if err != nil {
+			return nil, fmt.Errorf("failed to GetCellInfoNames: %w", err)
+		}
+
+		names = resp.Names
+	}
+
+	namesOnly := req.NamesOnly
+	if namesOnly {
```

Comment:
Just a question about """idiomatic go""": isn't this the same as `if namesOnly && len(req.Cells) != 0`?

Reply:
yuppppp (i originally had An Thought about doing the early return namesOnly loop in this section too, but never came back here), thanks!